src/platforms/android/androidappimageprovider.cpp

```diff
@@ -19,8 +18,16 @@ Logger logger(LOG_CONTROLLER, "AndroidAppImageProvider");
 AndroidAppImageProvider::AndroidAppImageProvider(QObject* parent)
     : QQuickImageProvider(QQuickImageProvider::Image,
                           QQmlImageProviderBase::ForceAsynchronousImageLoading),
-      QObject(parent) {
+#if QT_VERSION < 0x060000
```

Comment:
This is okay for as long as Qt6 is in beta, but once its properly released, it would be acceptable to assume a specific version of Qt for any source files that are Android-specific and to drop support for Qt5 in order to clean this up.

Reply:
This code has been moved to a separate class, but the approach has not changed: we need to treat the parent object differently between Qt5 and Qt6.